x-pack/plugins/secops/public/components/page/network/ip_overview/field_renderers.tsx

```diff
@@ -0,0 +1,138 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { getOr, isString } from 'lodash/fp';
+import React from 'react';
+
+import {
+  AutonomousSystem,
+  HostEcsFields,
+  IpOverviewData,
+  IpOverviewType,
+  Overview,
+} from '../../../../graphql/types';
+import { DefaultDraggable } from '../../../draggables';
+import { getEmptyTagValue } from '../../../empty_value';
+import { ExternalLinkIcon } from '../../../external_link_icon';
+import { PreferenceFormattedDate } from '../../../formatted_date';
+import { HostDetailsLink, ReputationLink, VirusTotalLink, WhoIsLink } from '../../../links';
+
+import { IpOverviewId } from './index';
+import * as i18n from './translations';
+
+export const locationRenderer = (
+  fieldNames: string[],
+  data: IpOverviewData
+): React.ReactElement => {
+  return fieldNames.length > 0 &&
+    fieldNames.every((fieldName: string) => getOr(null, fieldName, data)) ? (
+    <EuiFlexGroup alignItems="center" gutterSize="none" data-test-subj="location-field">
+      {fieldNames.map((fieldName: string, index: number) => {
+        const locationValue = getOr('', fieldName, data);
+        return (
+          <React.Fragment key={`${IpOverviewId}-${fieldName}`}>
+            {index ? ',\u00A0' : ''}
+            <EuiFlexItem grow={false}>
+              <DefaultDraggable
+                id={`${IpOverviewId}-${fieldName}`}
+                field={fieldName}
+                value={locationValue}
+              />
+            </EuiFlexItem>
+          </React.Fragment>
+        );
+      })}
+    </EuiFlexGroup>
+  ) : (
+    getEmptyTagValue()
+  );
+};
+
+export const dateRenderer = (fieldName: string, data: Overview): React.ReactElement => {
+  const value = getOr(null, fieldName, data);
+  return value && isString(value) ? (
+    <PreferenceFormattedDate value={new Date(value)} />
+  ) : (
+    getEmptyTagValue()
+  );
+};
+
+export const autonomousSystemRenderer = (
+  as: AutonomousSystem,
+  flowType: IpOverviewType
+): React.ReactElement => {
+  return as && as.as_org && as.asn ? (
+    <EuiFlexGroup alignItems="center" gutterSize="none">
+      <EuiFlexItem grow={false}>
+        <DefaultDraggable
+          id={`${IpOverviewId}-${flowType}.autonomous_system.as_org`}
+          field={`${flowType}.autonomous_system.as_org`}
+          value={as.as_org}
+        />{' '}
+        /
+        <DefaultDraggable
+          id={`${IpOverviewId}-${flowType}.autonomous_system.asn`}
+          field={`${flowType}.autonomous_system.asn`}
+          value={as.asn}
+        />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  ) : (
+    getEmptyTagValue()
+  );
+};
+
+export const hostIdRenderer = (host: HostEcsFields, ipFilter: string): React.ReactElement => {
+  return host.id && host.ip && host.ip.includes(ipFilter) ? (
```

Comment:
You can use implicit values if you want to:

```ts
export const hostIdRenderer = (host: HostEcsFields, ipFilter: string): React.ReactElement =>
  host.id && host.ip && host.ip.includes(ipFilter) ? (
    <EuiFlexGroup alignItems="center" gutterSize="none">
      <EuiFlexItem grow={false}>
        <DefaultDraggable id={`${IpOverviewId}-host-id`} field={'host.id'} value={host.id}>
          <HostDetailsLink hostId={host.id} />
        </DefaultDraggable>
      </EuiFlexItem>
    </EuiFlexGroup>
  ) : (
    getEmptyTagValue()
  );
```

It looks readable either way, just pointing it out if you enjoy the implicit's with prettier or not.

Reply:
Thanks for the heads up! I usually start with an explicit return if I don't know quite what the function is going to look like, but these are all pretty basic so I think an implicit return works well here too.